src/promptflow/tests/sdk_cli_test/recording_utilities/mock_tool.py

```diff
@@ -0,0 +1,194 @@
+import functools
```

Comment:
maybe add some comments for these functions, if you will do an update

Reply:
OKay